pyroute2/netlink/nl80211/__init__.py

```diff
@@ -1216,44 +1216,97 @@ def decode(self):
                 self.value["TDLS_PEER"] = False
                 self.value["ASSOCIATED"] = False
 
-                init = offset = self.offset + 4
-                while (offset - init) < (self.length - 4):
-                    (msg_type, length) = struct.unpack_from(
-                        'BB', self.data, offset
-                    )
-                    mask, set_ = struct.unpack_from(
-                        'II', self.data, offset + 2
-                    )
+                offset = self.offset + 4
+                mask, set_ = struct.unpack_from(
+                    'II', self.data, offset
+                )
 
-                    if mask & NL80211_STA_FLAG_AUTHORIZED:
-                        if set_ & NL80211_STA_FLAG_AUTHORIZED:
-                            self.value["AUTHORIZED"] = True
+                if mask & (1 << NL80211_STA_FLAG_AUTHORIZED):
```

Comment:
Could it be better to move this shift to the flag definition like it is done for `NL80211_SCAN_FLAG_LOW_PRIORITY` etc?